error.go

```diff
@@ -44,6 +44,19 @@ const (
 	ErrCodeUnknown
 )
 
+// ErrInvalidEndpoint indicates a endpoint error in the marathon urls
+type ErrInvalidEndpoint struct {
```

Comment:
Conventionally, error types end in `Error`, so this one should be called `InvalidEndpointError` ([reference](https://github.com/golang/go/wiki/Errors#naming)).

Would make `newInvalidEndpointError` fit more consistently as well.